### ISSUE TITLE
EndOfStreamError: Reached End Of Stream with Ember Websockets

### ISSUE THREAD

**Issue by CharlesAHunt** [http4s/http4s#5892]
Issue occurs with Ember only, switching back to Blaze until resolved.

Websockets encounter the following error on Ember server using `withHttpWebSocketApp(...)`

```
org.http4s.ember.server.EmberServerBuilderCompanionPlatform   WebSocket connection terminated with exception
org.http4s.ember.server.internal.WebSocketHelpers$EndOfStreamError: Reached End Of Stream
org.http4s.ember.server.internal.WebSocketHelpers$.$anonfun$decodeFrames$2(WebSocketHelpers.scala:224)
```

The error seems to be with the case None => Pull.raiseError(EndOfStreamError()).

It seems the new Ember implementation is interpreting no chunks/frames on the socket as an error, when it should continue waiting for a Close frame or a ping timeout.  Unfortunately, this is causing all my healthy clients to get their socket connections closed as soon as I switched from Blaze to Ember.

If this functionality is needed for some reason, can we add a configuration option to the EmberServerBuilder?  Maybe a `.waitForMoreChunks(true)` to override this error getting thrown?

**Comment by ChristopherDavenport** [comment]
None there indicates that the pipe has closed. And no more bytes will be received. Can you provide some more details on what was happening when these occured?

**Comment by CharlesAHunt** [comment]
@ChristopherDavenport Thanks for the response.  Turns out this is just a new error I hadn't seen before in the Ember implementation that was not the root cause of the issue and is a normal expected error when the Server shuts down socket connections.

I did discover the root cause of our issue, and it's that the Ember server is not passing Pings on from `handleIncomingFrames` in the following case:

`case WebSocketFrame.Ping(data) => writeFrame(WebSocketFrame.Pong(data)).as(None)`

We do have `.withFilterPingPongs(false)` set in the WebSocketBuilder2, so this case above not returning the Some(Ping) effectively makes the withFilterPingPongs broken.  (Our implementation relies on the Pings to log and set a health status which some critical behavior depends on)

I'll close this issue and create a separate issue and a PR to resolve this Ping filtering problem.

Thanks again @ChristopherDavenport !
